crates/router/src/routes/api_keys.rs

```diff
@@ -335,3 +337,34 @@ pub async fn api_key_list(
     )
     .await
 }
+#[cfg(feature = "v2")]
+#[instrument(skip_all, fields(flow = ?Flow::ApiKeyList))]
+pub async fn api_key_list(
+    state: web::Data<AppState>,
+    req: HttpRequest,
+    query: web::Query<api_types::ListApiKeyConstraints>,
+) -> impl Responder {
+    let flow = Flow::ApiKeyList;
+    let payload = query.into_inner();
+
+    api::server_wrap(
+        flow,
+        state,
+        &req,
+        (payload.limit, payload.skip),
+        |state, authentication_data, (limit, offset), _| async move {
+            let merchant_id = authentication_data.merchant_account.get_id().to_owned();
+            api_keys::list_api_keys(state, merchant_id, limit, offset).await
```

Comment:
You can directly pass `payload` here.
```suggestion
        payload,
        |state, authentication_data, payload, _| async move {
            let merchant_id = authentication_data.merchant_account.get_id().to_owned();
            api_keys::list_api_keys(state, merchant_id, payload.limit, payload.skip).await
```